### ISSUE TITLE
4.3.0 has back-incompatible changes

### ISSUE THREAD

**Issue by tpoterba** [micheles/decorator#48]
The version 4.3.0 removes `getargspec` from the module, which broke some scripts that worked with 4.2.1. We've resolved our issues by relying on a specific snapshot instead of 4.x, but I imagine there are others who might be affected as well. 

Was this change intentional? Semantic versioning dictates that minor versions are back-compatible, so it might be best to add it back until a 5.x version.

**Comment by micheles** [comment]
Yes, it was intentional. `getargspec` is not a major feature of the decorator module and it felt okay to remove it, this is why the release number is 4.3.0 and no 4.2.2. Notice that `getargspec` has been deprecated in Python for years and even in the decorator module was provided just for historical reasons (i.e. I did not make the work to remove it before). You should use `getfullargspec` instead.
If other users will protest, I may consider restoring it in the future, but it should not be too difficult for you to change your scripts,a simple search and replace should suffice.
